application/src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/transit/RemoveTransitIfStreetOnlyIsBetter.java

```diff
@@ -34,19 +59,36 @@ public String name() {
   @Override
   public List<Itinerary> flagForRemoval(List<Itinerary> itineraries) {
     // Find the best street-all-the-way option
-    OptionalInt minStreetCost = itineraries
+    OptionalInt minStreetCostOption = itineraries
       .stream()
       .filter(Itinerary::isStreetOnly)
       .mapToInt(Itinerary::generalizedCost)
       .min();
+    Cost minStreetCost = null;
 
-    if (minStreetCost.isEmpty()) {
+    if (minStreetCostOption.isEmpty() && generalizedCostMaxLimit == null) {
+      // If no cost is found an empty list is returned.
       return List.of();
+    } else if (minStreetCostOption.isPresent() && generalizedCostMaxLimit != null) {
+      // This case should not be possible.
+      throw new UnsupportedOperationException(
```

Comment:
I think this is probably not the correct exception. Perhaps IllegalStateException.